src/transformers/models/seamless_m4t/feature_extraction_seamless_m4t.py

```diff
@@ -251,12 +282,12 @@ def __call__(
         elif isinstance(raw_speech, np.ndarray) and raw_speech.dtype is np.dtype(np.float64):
             raw_speech = raw_speech.astype(np.float32)
 
-        # always return batch
-        if not is_batched:
-            raw_speech = [raw_speech]
-
-        # extract fbank features
-        features = [self._extract_fbank_features(waveform) for waveform in raw_speech]
+        # Extract fbank features, ensuring the result is always a batch
+        features = (
+            [self._extract_fbank_features(raw_speech)]
+            if not is_batched
+            else self._extract_fbank_features_batch(raw_speech)
+        )
```

Comment:
Hey @ravenouse, thanks for continuing the great work here!

Quick question on my side, why don't we use `_extract_fbank_features_batch` in any case, since we add the following code snippet that created a batch anyway:

```py
        # always return batch
        if not is_batched:
            raw_speech = [raw_speech]
```

This should be okay right, given that `spectrogram_batch` works when BS = 1?

Thanks in advance!

Reply:
In any case, this is quite convincing. Let's operate as you've done

From which batch size do we see an improvement?